### PR TITLE
Zypper patch system before test start on PC

### PR DESCRIPTION
**Pull request by lilyeyes**
Fully patch system before test start on public cloud qesap-deployment test cases
TEAM-8437 - [qesap-deployment] Fully patch system before test start

1. Fully patch system before test start via ansible playbook
2. Fix the timed out issue introduce in this "fully patch"
3. Remove previous workaround on "zypper up sudo" 

- Related ticket: https://jira.suse.com/browse/TEAM-8437
- Needles: NA
- Verification run:
 1. sle-15-SP2-HanaSr-Azure-Byos: hanasr_azure_test
  https://openqaworker15.qa.suse.cz/tests/229811   (passed, (02:02 hours))
 2. sle-15-SP4-HanaSr-Aws-Byos: hanasr_aws_test_saptune
  http://openqaworker15.qa.suse.cz/t229807 (Command "sudo SAPHanaSR-showAttr" passed)
  (The "Fenced database 'vmhana01' is not offline" failure is a known issue)  
 3. sle-15-SP4-Azure-SAP-BYOS: SAPHanaSR-ScaleUp-PerfOpt
  https://openqaworker15.qa.suse.cz/tests/229815   (passed)
4. sle-15-SP5-Qesap-Azure-Byos: qesap_azure_sapconf_test
  https://openqaworker15.qa.suse.cz/tests/231114 (passed, VR for code changes in "config.yaml" files)

### CODE REVIEW

File: data/sles4sap/qe_sap_deployment/qesap_aws.yaml [hunk 1]
```diff
@@ -36,6 +36,7 @@ ansible:
     secondary_site: 'miky'
   create:
     - registration.yaml -e reg_code='%SCC_REGCODE_SLES4SAP%' -e email_address=testing@suse.com
+    - fully-patch-system.yaml
```
Comment (mpagot) [inline data/sles4sap/qe_sap_deployment/qesap_aws.yaml:39]: I'm not sure we want all qesap regression to use this

File: data/sles4sap/qe_sap_deployment/trento_azure.yaml [hunk 2]
```diff
@@ -32,6 +32,7 @@ ansible:
     secondary_site: 'miky'
   create:
     - registration.yaml -e reg_code='%SCC_REGCODE_SLES4SAP%' -e email_address='testing@suse.com'
+    - fully-patch-system.yaml
```
Comment (mpagot) [inline data/sles4sap/qe_sap_deployment/trento_azure.yaml:35]: I'm not sure Trento want this

File: tests/sles4sap/publiccloud/hana_sr_takeover.pm [hunk 3]
```diff
@@ -35,7 +35,7 @@ sub run {
     $self->{my_instance} = $target_site;
 
     # Check initial cluster status
-    $self->run_cmd(cmd => 'zypper -n in ClusterTools2');
+    $self->run_cmd(cmd => 'zypper -n in ClusterTools2', timeout => 300);
```
Comment (mpagot) [inline tests/sles4sap/publiccloud/hana_sr_takeover.pm:38]: is it related to full update? Can this be moved in a dedicated PR?

File: data/sles4sap/qe_sap_deployment/trento_azure.yaml (hunk 2 above)
Comment (lilyeyes) [reply data/sles4sap/qe_sap_deployment/trento_azure.yaml:35]: Not sure too, but from the "Description" of this ticket: https://jira.suse.com/browse/TEAM-8437 I think I need to apply full patch for as many as test cases.

File: data/sles4sap/qe_sap_deployment/qesap_aws.yaml (hunk 1 above)
Comment (lilyeyes) [reply data/sles4sap/qe_sap_deployment/qesap_aws.yaml:39]: Not sure too.

Comment (alvarocarvajald) [reply data/sles4sap/qe_sap_deployment/qesap_aws.yaml:39]: I'd say either remove it or leave it depending on what would improve the stability of the qesap regression tests. Removing the `fully-patch-system.yaml` would result in a quicker test, but if by some change the unpatched system is broken and a `zypper patch` would fix the issue then it would be preferable. I'm somewhat leaning to removing it, but I'm not 100% sure.

File: data/sles4sap/qe_sap_deployment/trento_azure.yaml (hunk 2 above)
Comment (alvarocarvajald) [reply data/sles4sap/qe_sap_deployment/trento_azure.yaml:35]: I think we should fully patch in trento scenario.

File: tests/sles4sap/publiccloud/hana_sr_takeover.pm (hunk 3 above)
Comment (alvarocarvajald) [reply tests/sles4sap/publiccloud/hana_sr_takeover.pm:38]: @lilyeyes perhaps what you can do is move the change into this file to its own commit so it can easily reverted if needed.

**Comment by alvarocarvajald** [comment]
LGTM. Only comment is regarding qesap regression tests.

File: data/sles4sap/qe_sap_deployment/qesap_aws.yaml (hunk 1 above)
Comment (mpagot) [reply data/sles4sap/qe_sap_deployment/qesap_aws.yaml:39]: > I'd say either remove it or leave it depending on what would improve the stability of the qesap regression tests. Removing the `fully-patch-system.yaml` would result in a quicker test, but if by some change the unpatched system is broken and a `zypper patch` would fix the issue then it would be preferable. I'm somewhat leaning to removing it, but I'm not 100% sure.

I like to have a place (test) to check if the full patch makes any difference. I like to have only one test using it. Or one test not using it.

Comment (lilyeyes) [reply data/sles4sap/qe_sap_deployment/qesap_aws.yaml:39]: Removed `fully-patch-system.yaml` from `data/sles4sap/qe_sap_deployment/qesap_aws.yaml`.

File: tests/sles4sap/publiccloud/hana_sr_takeover.pm (hunk 3 above)
Comment (lilyeyes) [reply tests/sles4sap/publiccloud/hana_sr_takeover.pm:38]: Done, submitted another commit in this PR.

**Review by alvarocarvajald — APPROVED** [review]
LGTM

**Comment by lilyeyes** [comment]
@mpagot  Please help to review again.

**Review by mpagot — APPROVED** [review]
LGTM, needs some action before to be committed:
-  qe-sap-deployment PR177 to be reverted
- qe-sap-deployment release 0.16.0
- JobGroup MR merged and deployed
